More templated policies docs (#19312)
hashicorp/consul: website/content/docs/security/acl/acl-roles.mdx

- `Policies`: Specifies a the list of policies that are applicable for the role. The object can reference the policy `ID` or `Name` attribute.
- `TemplatedPolicies`: Specifies a list of templated polcicies that are applicable for the role. See [Templated Policies](#templated-policies) for details.
- `ServiceIdentities`: Specifies a list of services that are applicable for the role. See [Service Identities](#service-identities) for details.
- `NodeIdentities`: Specifies a list of nodes that are applicable for the role. See [Node Identities](#node-identities) for details.
- `Namespace`: <EnterpriseAlert inline /> The namespace that the policy resides in. Roles can only be linked to policies that are defined in the same namespace. See [Namespaces](/consul/docs/enterprise/namespaces) for additional information. Requires Consul Enterprise 1.7.0+
- `Partition`: <EnterpriseAlert inline/> The admin partition that the policy resides in. Roles can only be linked to policies that are defined in the same admin partition. See [Admin Partitions](/consul/docs/enterprise/admin-partitions) for additional information. Requires Consul Enterprise 1.10.0+.

## Templated policies

> You can specify a templated policy when configuring roles or linking tokens to policies. Templated policies enable you to quickly construct policies for common consul use cases, rather than creating identical policies for each use cases.

Reviewer: ```suggestion
You can specify a templated policy when configuring roles or linking tokens to policies. Templated policies enable you to quickly construct policies for common Consul use cases, rather than creating identical policies for each use cases.
```

> Templated policies are used during the authorization process to automatically generate a policy for the use case specified. Consul links the generated policy to the role or token so that it will have permission for the specific use case.  

Reviewer: ```suggestion
Consul uses templated policies during the authorization process to automatically generate a policy for the use case specified. Consul links the generated policy to the role or token so that it will have permission for the specific use case.  
```

### Templated policy specification

> Use the following syntax to define a templated policy:

Reviewer: ```suggestion
The following templated policy example configuration ____________
```

explain what this example does

```json
{
  "TemplatedPolicies": [
    {
      "TemplateName": "builtin/service",
      "TemplateVariables": {
        "Name": "api"
      }
    }
  ]
}
```

- `TemplatedPolicies`: Declares a templated policy block.
- `TemplatedPolicies.TemplateName`: String value that specifies the name of the templated policy you want to use.
- `TemplatedPolicies.TemplateVariables`: Map that specifies the required variables for the templated policies. This field is optional as not all templated policies require variables.

Refer to the [API documentation for roles](/consul/api-docs/acl/roles#sample-payload) for additional information and examples.

> -> **Scope for Namespace and Admin Partition** - In Consul Enterprise, templated policies inherit the namespace or admin partition scope of the corresponding ACL token or role.

Reviewer: ```suggestion
In Consul Enterprise, templated policies inherit the namespace or admin partition scope of the corresponding ACL token or role.
```

> The following policy is generated for the `builtin/service` templated policy for a service named `api`:

Reviewer: ```suggestion
The `builtin/service` sample templated policy generates the following policy for a service named `api`:
```

```hcl
# Allow the service and its sidecar proxy to register into the catalog.
service "api" {
    policy = "write"
}
service "api-sidecar-proxy" {
    policy = "write"
}

# Allow for any potential upstreams to be resolved.
service_prefix "" {
    policy = "read"
}
node_prefix "" {
    policy = "read"
}
```

Refer to the [rules reference](/consul/docs/security/acl/acl-rules) for information about the rules in the policy.

### Example

> The following role configuration contains templated policies for that give the role required permission for a service named `web` to register itself and its sidecar into the catalog.

Reviewer: ```suggestion
The following role configuration contains a templated policy that gives the role required permission for a service named `web` to register itself and its sidecar into the catalog.
```

this sentence doesn't super make sense, can you rephrase it?

<CodeBlockConfig filename="example-role.json">

```json
{
  "Name": "example-role",
  "Description": "Showcases all input parameters",
  "TemplatedPolicies": [
    {
      "TemplateName": "builtin/service",
      "TemplateVariables": {
        "Name": "web"
      }
    }
  ]
}
```

</CodeBlockConfig>

> During the authorization process, the following policies for the `web` services will be generated and linked to the token:

Reviewer: ```suggestion
During the authorization process, Consul generates the following policies for the `web` services and links it to the token:
```

active voice, now it's clear what generates and links the policies
